frontend/src/plugins/auth.js

```diff
@@ -113,10 +113,20 @@ async function redirectToOAuthLogin(provider) {
   return apiStore
     .href(apiStore.get(), provider, { callback: encodeURI(returnUrl) })
     .then((url) => {
-      window.location.href = window.environment.API_ROOT_URL + url
+      const apiRootUrl = window.environment.API_ROOT_URL
+      window.location.href = (hasHost(apiRootUrl) ? apiRootUrl : '') + url
```

Comment:
This seems very much specific to our current setup of "either all services live on a subdomain with no path prefix, or they all live on the same domain with different path prefixes". In the rest of this PR, I was under the impression that everything is generalized so that services may live on subdomains or have path prefixes, or both.
Isn't this more or less a bug in hal-json-vuex? I feel like hal-json-vuex should either return absolute URIs, or cut off the path prefix from all URIs. What is your opinion on this?

Reply:
hal-json-vuex does everything right i think.
We have a mixture between using hal-json-vuex and the browser.
Haljsonvuex uses relative paths from apiRootUrl, and adds the prefix before the requests, but window.location.href does not know about this.
Thus the logic to put together urls is duplicated.

Before #3262 , we still use an api url with a different host, so we have to include the host when putting together the url.
After (or in) #3262, we can remove this and use relative url's everywhere.